qiita_db/test/test_analysis.py

```diff
@@ -427,6 +428,26 @@ def test_build_files_raises_value_error(self):
         with self.assertRaises(ValueError):
             self.analysis.build_files(-10)
 
+    def test_add_file(self):
+        fp = join(get_mountpoint('analysis')[0][1], 'testfile.txt')
+        try:
```

Comment:
If I understand correctly: if you can write to the file path you add it to the analysis and regardless of the result of these two lines you check if it exists and if it exists you remove it? Should that fp be removed in the teardown method of the test class? So we can guarantee that it is removed, the finally block here is kinda confusing.